### DISCUSSION TITLE
How to extract the query param values

### DISCUSSION THREAD

**Discussion by RajuJosephK** [mockoon/mockoon#441] [Help] [answered]
**I am making the below request**
    http://127.0.0.1:3001/resourceful/entity/client-v4?quicksearch=ips:TEST1234&fields.0=John

**In Mockoon I have setup to return the below response body**
      {
      "quick_search": "{{ queryParam 'quicksearch'}}",
      "fields": "{{ queryParam 'fields.0'}}"
      }

**I get the response** 
    {
    "quick_search": "ips:TEST1234",
    "fields": ""
    }

**I am expecting the value "JOHN" in the response but its not returning.  How can I achieve this.**

**Comment by 255kb** [comment, marked as answer]
Dots inside the `queryparam` helper `{{ queryParam 'fields.0'}}` will directly get translated to an object path (we are using [object-path library](https://www.npmjs.com/package/object-path)), so, using dots as a property name `fields.0` won't work. 
Also, the usual way of passing arrays in query params is `?prop[]=value1&prop[]=value2`, at least it's the one that [qs library](https://www.npmjs.com/package/qs) we are using, understands. 
So you may want to replace your query param by either `http://127.0.0.1:3001/resourceful/entity/client-v4?quicksearch=ips:TEST1234&fields[]=John` or `http://127.0.0.1:3001/resourceful/entity/client-v4?quicksearch=ips:TEST1234&fields[0]=John`.

**Reply by RajuJosephK** [reply]
Thanks

**Reply by RajuJosephK** [reply]
As mentioned, the the way query params being passed is incorrect.  Will replace the query params.  That should work.  Answers my query.
